src/components/ChatApp/TopBar.react.js

```diff
@@ -70,9 +78,13 @@ class TopBar extends Component {
 				anchorOrigin={{ horizontal: 'right', vertical: 'top' }}
 			>
 				<MenuItem primaryText="Settings"
-					onClick={this.props.handleSettings} />
-				<MenuItem primaryText="Login" onTouchTap={this.props.handleOpen} />
-				<MenuItem primaryText="Sign Up" onTouchTap={this.props.handleSignUp} />
+				onClick={this.props.handleSettings}
+				rightIcon={<Settings/>} />
+				<MenuItem primaryText="Login" onTouchTap={this.props.handleOpen}
+				 />
+				<MenuItem primaryText="Sign Up"
+				containerElement={<Link to="/signup" />}
```

Comment:
Please use `onTouchTap={this.props.handleSignUp}` instead of `Link to` , we removed the `/signup` route